src/state/actions/dispatchANTInteraction.ts

```diff
@@ -50,6 +43,15 @@ export default async function dispatchANTInteraction({
   ao: AoClient;
   stepCallback?: (step?: Record<string, string> | string) => Promise<void>;
 }): Promise<ContractInteraction> {
+  stepCallback ??= async (step) => {
```

Comment:
does this need to be `async`?

Reply:
To satisfy the type, technically no, i could manually construct and return a promise, but making it async is an easier shorthand.

The reason its async is to allow the callback to be suspended and another operation to be performed/predicated by the calling workflow - such as UI input.